commands.c

```diff
@@ -942,83 +942,79 @@ void mutt_display_address(struct Envelope *env)
 
 /**
  * set_copy_flags - Set the flags for a message copy
- * @param[in]  e       Email
- * @param[in]  decode  If true, decode the message
- * @param[in]  decrypt If true, decrypt the message
- * @param[out] cmflags Flags, see #CopyMessageFlags
- * @param[out] chflags Flags, see #CopyHeaderFlags
+ * @param[in]  e               Email
+ * @param[in]  transform_opt   decode/decrypt/none
+ * @param[out] cmflags         Flags, see #CopyMessageFlags
+ * @param[out] chflags         Flags, see #CopyHeaderFlags
  */
-static void set_copy_flags(struct Email *e, bool decode, bool decrypt,
+static void set_copy_flags(struct Email *e, enum MessageTransformOpt transform_opt,
                            CopyMessageFlags *cmflags, CopyHeaderFlags *chflags)
 {
   *cmflags = MUTT_CM_NO_FLAGS;
   *chflags = CH_UPDATE_LEN;
 
-  if ((WithCrypto != 0) && !decode && decrypt && (e->security & SEC_ENCRYPT))
+  const bool needs_decrypt = 
+      (transform_opt == TRANSFORM_DECRYPT) && 
+      (e->security & SEC_ENCRYPT);
+
+  const bool want_pgp   = WithCrypto & APPLICATION_PGP;
+  const bool want_smime = WithCrypto & APPLICATION_SMIME;
+  const bool is_pgp     = mutt_is_application_pgp(e->body)   & SEC_ENCRYPT;
+  const bool is_smime   = mutt_is_application_smime(e->body) & SEC_ENCRYPT;
+
```

Comment:
I should probably have used 3rd or 1st person consistently.. s/want/wants/ or s/needs/need/ :)